tests/integration/aws_s3/test_aws_s3.py

```diff
@@ -60,73 +57,101 @@ def aws_s3_configured(extra_params=None):
-        res = {
-            "bucket_path": bucket_path,
-            "object_path": object_path,
-            "df_path": df_path,
-            "object_url": object_url,
-            "df_url": f"{bucket_path}/{df_path}",
-            "blob_url": f"{object_url}.blob",
-            "parquet_url": f"{object_url}.parquet",
-        }
-        return res
-
-    def setup_method(self, method):
-        self._bucket_name = config["env"].get("bucket_name")
-        self._access_key_id = config["env"].get("AWS_ACCESS_KEY_ID")
-        self._secret_access_key = config["env"].get("AWS_SECRET_ACCESS_KEY")
-
-        object_dir = "test_mlrun_s3_objects"
-        object_file = f"file_{random.randint(0, 1000)}.txt"
-        csv_file = f"file_{random.randint(0,1000)}.csv"
-
-        self.s3 = {}
-        self.s3["s3"] = self._make_target_names(
-            "s3://", self._bucket_name, object_dir, object_file, csv_file
+    @classmethod
+    def setup_class(cls):
+        cls.assets_path = os.path.join(
+            os.path.dirname(os.path.dirname(__file__)), "assets"
         )
-        self.s3["ds"] = self._make_target_names(
-            "ds://s3ds_profile/", self._bucket_name, object_dir, object_file, csv_file
+        cls.env = config["env"]
+        cls._bucket_name = cls.env.get("bucket_name")
+        cls._access_key_id = cls.env.get("AWS_ACCESS_KEY_ID")
+        cls._secret_access_key = cls.env.get("AWS_SECRET_ACCESS_KEY")
+        cls.profile_name = "s3ds_profile"
+        cls.test_dir = "/test_mlrun_s3"
+        cls.run_dir = cls.test_dir + f"/run_{uuid.uuid4()}"
+        cls.test_file = os.path.join(cls.assets_path, "test.txt")
+        with open(cls.test_file) as f:
+            cls.test_string = f.read()
+        cls._fs = fsspec.filesystem(
+            "s3", anon=False, key=cls._access_key_id, secret=cls._secret_access_key
         )
-        profile = DatastoreProfileS3(
-            name="s3ds_profile",
+
+    @classmethod
+    def teardown_class(cls):
+        test_dir = f"{cls._bucket_name}{cls.test_dir}"
+        if not cls._fs:
+            return
+        if cls._fs.exists(test_dir):
+            cls._fs.delete(test_dir, recursive=True)
+            logger.debug("test directory has been deleted.")
+
+    def setup_method(self, method):
+        store_manager.reset_secrets()
+        self.profile = DatastoreProfileS3(
+            name=self.profile_name,
             access_key_id=self._access_key_id,
             secret_key=self._secret_access_key,
         )
-        register_temporary_client_datastore_profile(profile)
+        register_temporary_client_datastore_profile(self.profile)
+
+    @pytest.fixture(autouse=True)
+    def setup_before_each_test(self, use_datastore_profile):
+        # We give priority to profiles, then to secrets, and finally to environment variables.
+        # We want to ensure that we test these priorities in the correct order.
+        mlrun.datastore.store_manager.reset_secrets()
+
```

Comment:
```suggestion
        mlrun.datastore.store_manager.reset_secrets()

        # We give priority to profiles, then to secrets, and finally to environment variables.
        # We want to ensure that we test these priorities in the correct order.
```
or alternatively remove the blank line. Otherwise, it's not obvious what code this comment is referring to.